extensions/web/08_text_encoding.js

```diff
@@ -4338,6 +4338,129 @@
     }
   }
 
+  class TextDecoderStream {
+    #decoder;
+    #transform;
+
+    get encoding() {
+      return this.#decoder.encoding;
+    }
+    get fatal() {
+      return this.#decoder.fatal;
+    }
+    get ignoreBOM() {
+      return this.#decoder.ignoreBOM;
+    }
+
+    constructor(label = "utf-8", options = { fatal: false }) {
+      this.#decoder = new TextDecoder(label, options);
+      this.#transform = new TransformStream({
+        // The transform and flush functions need access to TextDecoderStream's
+        // `this`, so they are defined as functions rather than methods.
+        transform: (chunk, controller) => {
+          try {
+            // Step 1 in "decode and enqueue a chunk" converts `chunk` to an
+            // `[AllowShared] BufferSource`, which is the same type as the first
+            // argument to TextDecoder.prototype.decode – except that that
+            // argument is marked as optional. Therefore, we need to
+            // special-case `undefined`.
+            if (chunk === undefined) {
+              throw new TypeError(
+                "Chunk in the stream is not of type ArrayBuffer or ArrayBufferView",
+              );
+            }
```

Comment:
Convert chunk to a buffer source with `webidl.converters.BufferSource`. For an example see `https://github.com/denoland/deno/blob/main/extensions/web/08_text_encoding.js#L75-L79`.